docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,221 @@
+│  • /r: toggle the status of the RAG pipeline.                                                                                     │
+│  • /md <int>: previous response in Markdown based on input, if passed 1 then previous, if 2 then second last response and so on.  │
+│  • /s filepath: save current session to filepath                                                                                  │
+│  • /l filepath: load filepath and start a new session                                                                             │
+│  • /L filepath: load filepath (permanently) and start a new session                                                               │
+│                                                                                                                                   │
+│ Press Alt (or Meta) and Enter or Esc Enter to end multiline input.                                                                │
+╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
+```
+
+### 2.6 RAG Chat Options
+As we stated in [3.1 Working Assumptions](#31-working-assumption), we will introduce new configuration options for the spceific `chat` command,
+but we'll use flags and environment variables for the options that come from the embedding ingestion pipeline command.
+
+| Configuration FQN | Description | Default Value | CLI Flag | Environment Variable |
+|-------------------|-------------|---------------|----------|----------------------|
+| chat.rag.enabled | Enable or disable the RAG pipeline. | `false` | `--rag` (boolean)| `ILAB_CHAT_RAG_ENABLED` |
+| chat.rag.retriever.top_k | The maximum number of documents to retrieve. | `10` | `--retriever-top-k` | `ILAB_CHAT_RAG_RETRIEVER_TOP_K` |
+| chat.rag.prompt | Prompt template for RAG-based queries. | Examples below | `--rag-prompt` | `ILAB_CHAT_RAG_PROMPT` |
+| | Vector DB implementation, one of: `milvuslite`, **TBD** | `milvuslite` | `--vectordb-type` | `ILAB_VECTORDB_TYPE` |
+| | Vector DB service URI. | `./rag-output.db` | `--vectordb-uri` | `ILAB_VECTORDB_URI` |
+| | Vector DB connection token. | | `--vectordb-token` | `ILAB_VECTORDB_TOKEN` |
+| | Vector DB connection username. | | `--vectordb-username` | `ILAB_VECTORDB_USERNAME` |
+| | Vector DB connection password. | | `--vectordb-password` | `ILAB_VECTORDB_PASSWORD` |
+
+Equivalent YAML document for the newly proposed options:
+```yaml
+chat:
+  rag:
+    enabled: false
+    retriever:
+      top_k: 10
+    prompt: |
+      Given the following information, answer the question.
+      Context:
+      {{context}}
+      Question: {{question}}
+      Answer:
+```
+
+### 2.7 References
+* [Haystack-DocumentSplitter](https://github.com/deepset-ai/haystack/blob/f0c3692cf2a86c69de8738d53af925500e8a5126/haystack/components/preprocessors/document_splitter.py#L55)
```

Comment:
I think probably the Haystack splitter will wind up getting dropped from the solution in favor of something Docling-based.